src/core/lib/transport/call_spine.h

```diff
@@ -248,13 +249,20 @@ class CallInitiator {
   auto PullServerTrailingMetadata() {
     GPR_DEBUG_ASSERT(GetContext<Activity>() == &spine_->party());
     return PrioritizedRace(
-        Map(spine_->server_trailing_metadata().receiver.Next(),
-            [spine = spine_](
-                NextResult<ServerMetadataHandle> md) -> ServerMetadataHandle {
-              GPR_ASSERT(md.has_value());
-              return std::move(*md);
+        Seq(spine_->server_trailing_metadata().receiver.Next(),
+            [spine = spine_](NextResult<ServerMetadataHandle> md) mutable {
+              return [md = std::move(md),
+                      spine]() mutable -> Poll<ServerMetadataHandle> {
+                if (!md.has_value()) return Pending{};
```

Comment:
When would the provided `md` not have any value ? Why do we returning Pending{} in that case ?

Reply:
added a comment